packages/opds-web-client/src/components/CatalogLink.tsx

```diff
@@ -38,16 +38,16 @@ export default class CatalogLink extends React.Component<CatalogLinkProps, void>
   };
 
   render(): JSX.Element {
-    let { collectionUrl, bookUrl } = this.props;
+    let { collectionUrl, bookUrl, ...rest } = this.props;
     collectionUrl = collectionUrl || null;
     bookUrl = bookUrl || null;
 
     let location = this.context.pathFor(collectionUrl, bookUrl);
 
     // remove props that Link won't recognize
-    let props = Object.assign({}, this.props);
-    delete props["collectionUrl"];
-    delete props["bookUrl"];
+    let props = Object.assign({}, rest);
+    // delete props["collectionUrl"];
+    // delete props["bookUrl"];
```

Comment:
I think this was to prevent console warnings from having props that don't belong on a link?

Reply:
Got it. They did not come up when doing this because the `rest` only has the needed props.

Reply:
Maybe you could use this approach to get rid of the `JSON.stringify`s in other places.